### ISSUE TITLE
Add API control to UI and Filter reset button 

### ISSUE THREAD

**Issue by javier-molina** [AtlasOfLivingAustralia/extended-data-model#68]
Similar to Biocache, add a UI control to get an API query 

![Screen Shot 2022-09-29 at 3 33 38 pm](https://user-images.githubusercontent.com/8326394/192947448-df7afd54-b0ee-4959-ae3c-8442fe694609.png)

How do you do that in GraphQL?

**Comment by qifeng-bai** [comment]
Add API controls to dataset, event and sites page

[Example URL](https://events-test.ala.org.au/graphql?queryId=f705da97c3a59272561a421c844073218cf8d4fb&strict=true&variables={"predicate":{"type":"and","predicates":[{"type":"and","predicates":[{"type":"in","key":"datasetKey","values":["dr9942"]},{"type":"or","predicates":[{"type":"equals","key":"year","value":"2008"},{"type":"equals","key":"year","value":"2012"}]}]}]},"limit":50,"offset":0})
Questions:
Is the queryID persistent or has a expiration? 
the default 'size/limit' is 20, do we need to allow users to download full records?


<img width="397" alt="Screen Shot 2022-10-28 at 10 49 44 am" src="https://user-images.githubusercontent.com/31238531/198418368-e9b1e2ca-9ea6-4745-94d1-2a6eecbe7580.png">

**Comment by MortenHofft** [comment]
Something to be weary of with this approach is that those IDs are not persistent. They only live in a temporary cache on the server. So if we for example restart graphQL, then the link will not work anymore.
So it will likely work when the user copy paste it, but it isn't something they can copy into e.g. a script and expect it to work.

I too would like to draw more attention to the APIs. So far I've been imagining that we added a consistent 3 options on all pages (search as well as detail pages) with information on 
* how to access through the APIs
* how to cite 
* and a what-is-this-thing-I-am-looking-at. 

But exactly how that would work and look I haven't thought through at all. I just liked the idea that those options were available everywhere at the same location.

<img width="1121" alt="Screenshot 2022-10-31 at 11 12 53" src="https://user-images.githubusercontent.com/6759267/198985208-108c312a-466a-4740-9c68-e7af8aa50f66.png">


In other words, I think it is a great idea, but if we want stable url links in all cases, then we need to either encode the query into the url (perfectly possible for smaller queries). Or have a place to register queries where they are stored for future usage.

**Comment by qifeng-bai** [comment]
@MortenHofft I agree you idea about "I just liked the idea that those options were available everywhere at the same location."

However, in our current Event app, we have 'Dataset' 'Event' and 'Site' tabs in 'Nav bar'.  Selecting different tabs will generate different queries.  Putting 'API' button on 'Nav bar' may cause some confusions. So that is why we attach the API button on each page
![Screen Shot 2022-11-01 at 9 24 56 am](https://user-images.githubusercontent.com/31238531/199121827-79b9aaac-72db-496f-ba57-ad9e61b908cf.png)

**Comment by djtfmartin** [comment]
+ 1 for @MortenHofft 3 consistent options on all pages idea. I think this will work for the Events app.

**Comment by javier-molina** [comment]
I have the same concern as Bai, there are three different queries so I think it would be unintuitive to have a single API control at the top that will produce different results depending of the tab you are in or disappear if you are for example in the Map or download tabs.

Do you @MortenHofft , @djtfmartin do you still think this is the best location?

**Comment by djtfmartin** [comment]
I think if the objective is to emulate what the biocache does, then we just support a single query.

Toggling between the 4 tabs in the biocache UI doesnt change the query visible from the API button (but the underlying query is changing).

<img width="1273" alt="Screen Shot 2022-11-02 at 7 20 31 am" src="https://user-images.githubusercontent.com/444897/199424987-1a89c3c4-0108-4646-930c-950cceba30cc.png">

**Comment by qifeng-bai** [comment]
@javier-molina  any suggestions?

1, Move API button up to the tab, Clear filter button added.
![Screen Shot 2022-12-02 at 1 54 05 pm](https://user-images.githubusercontent.com/31238531/205204722-b118fc14-ec1b-4b33-9bdf-5d40eff6a3c3.png)




2, Use  the sidebar to display GraphQL details
![Screen Shot 2022-12-02 at 1 54 15 pm](https://user-images.githubusercontent.com/31238531/205204640-8696fd8c-ffc0-4f5a-91c4-59056f80f1e5.png)

**Comment by qifeng-bai** [comment]
Add enable GraphQL API and Filter reset buttons into siteConfig
https://github.com/gbif/gbif-web/commit/96b969530cd6c9c10291ff281cac0af262556cca

**Comment by qifeng-bai** [comment]
Those two buttons are not displayed by default.
They can be turn on/off in SiteConfig

PR: https://github.com/gbif/gbif-web/pull/299
